flows/interfaces.go

```diff
@@ -163,12 +163,17 @@ type Context interface {
 }
 
 type Event interface {
-	CreatedOn() *time.Time
+	CreatedOn() time.Time
 	SetCreatedOn(time.Time)
 
 	Step() StepUUID
 	SetStep(StepUUID)
 
+	FromCaller() bool
+	SetFromCaller(bool)
```

Comment:
Well now that I've embraced the "events as real history" view I'm cool with `created_on` being the real world time. Regarding step_uuid I think we could ditch it and infer it from the nested events in the path.. but for what we're trying to do currently creating flowsteps, it's kinda awkward to figure out which events in the session are which events in the path.

I was wondering about changing `events` to something like:

```
"new_events": [
  {
    "step_uuid": "1234..."
    "events": [
       ...
    ],
  }
]
```

That makes it easy to get stuff like step time, node uuid etc for each of your new events that you need to apply in the caller.